x-pack/filebeat/input/httpjson/value_tpl.go

```diff
@@ -40,6 +41,11 @@ const (
 var (
 	errEmptyTemplateResult = errors.New("the template result is empty")
 	errExecutingTemplate   = errors.New("the template execution failed")
+	errMinWrongNoOfArgs    = errors.New("template: :1:2: executing \"\" at <min>: wrong number of args for min: want at least 1 got 0")
+	errMaxWrongNoOfArgs    = errors.New("template: :1:2: executing \"\" at <max>: wrong number of args for max: want at least 1 got 0")
+	errMinUnknownType      = errors.New("template: :1:2: executing \"\" at <min 1 \"b\" -2>: error calling min: unknown type for \"b\" (string)")
+	errMaxUnknownType      = errors.New("template: :1:2: executing \"\" at <max 1 \"b\" -2>: error calling max: unknown type for \"b\" (string)")
+	errStrUnknownType      = "unknown type for %q (%T)"
```

Comment:
Rather than factoring out at this level, make a helper function that returns the right kind of error. Naked format strings are a bit of smell, particularly at this distance from the call site.